dependency/nomad_secure_variables_get_test.go

```diff
@@ -0,0 +1,294 @@
+package dependency
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	nomadapi "github.com/hashicorp/nomad/api"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewSVGetQuery(t *testing.T) {
+
+	cases := []struct {
+		name string
+		i    string
+		exp  *SVGetQuery
+		err  bool
+	}{
+		{
+			"empty",
+			"",
+			&SVGetQuery{},
+			false,
+		},
+
+		{
+			"dc_only",
+			"@dc1",
+			nil,
+			true,
+		},
+		{
+			"path",
+			"path",
+			&SVGetQuery{
+				path: "path",
+			},
+			false,
+		},
+		{
+			"dots",
+			"path.with.dots",
+			&SVGetQuery{
+				path: "path.with.dots",
+			},
+			false,
+		},
+		{
+			"slashes",
+			"path/with/slashes",
+			&SVGetQuery{
+				path: "path/with/slashes",
+			},
+			false,
+		},
+		{
+			"dashes",
+			"path-with-dashes",
+			&SVGetQuery{
+				path: "path-with-dashes",
+			},
+			false,
+		},
+		{
+			"leading_slash",
+			"/leading/slash",
+			&SVGetQuery{
+				path: "leading/slash",
+			},
+			false,
+		},
+		{
+			"trailing_slash",
+			"trailing/slash/",
+			&SVGetQuery{
+				path: "trailing/slash/",
+			},
+			false,
+		},
+		{
+			"underscores",
+			"path_with_underscores",
+			&SVGetQuery{
+				path: "path_with_underscores",
+			},
+			false,
+		},
+		{
+			"special_characters",
+			"config/facet:größe-lf-si",
+			&SVGetQuery{
+				path: "config/facet:größe-lf-si",
+			},
+			false,
+		},
+		{
+			"splat",
+			"config/*/timeouts/",
+			&SVGetQuery{
+				path: "config/*/timeouts/",
+			},
+			false,
+		},
+	}
+
+	for i, tc := range cases {
+		t.Run(fmt.Sprintf("%d_%s", i, tc.name), func(t *testing.T) {
+			act, err := NewSVGetQuery(tc.i)
+			if (err != nil) != tc.err {
+				t.Fatal(err)
+			}
+
+			if act != nil {
+				act.stopCh = nil
+			}
+
+			assert.Equal(t, tc.exp, act)
+		})
+	}
+	fmt.Println("done")
+}
+
+func TestSVGetQuery_Fetch(t *testing.T) {
+
+	type svmap map[string]string
+	_ = testNomad.CreateSecureVariable("test-kv-get/path", svmap{"bar": "barp"})
```

Comment:
Rather than `fmt.Printf`'ing in the test helper, we should probably return the error here and `t.Fatal` (or `require.NoError`, as we have testify in this repo); there's no reasonable behavior for the test after this point.